### ISSUE TITLE
Using auto-accessor with `readonly` or `declare` should raise an error

### ISSUE THREAD

**Issue by fwienber** [microsoft/TypeScript#51820]
# Bug Report

### 🔎 Search Terms

auto accessor readonly declare

### 🕗 Version & Regression Information

Version 4.9.3

- I was unable to test this on prior versions because the auto-accessor feature is new in 4.9

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8BQ0PTMYBTK8ATtIbmACbwB2IAntAGaIBc0VArgLYBGuxAvNACsAbnSZseAsXK5QYUtG4LWEAC6EAllQDmYgL6pUwauuidawaIKq4A7nEQAKAJRiLwAHRMkggGxiJlQQ8CC4niDwOk4e3ohuqEA)

### 💻 Code

```ts
class Foo {
    accessor readonly foo: number = 5;
    //       ^ no error here
    accessor declare bar: string;
    //       ^ no error here
}

const myc = new Foo();
myc.foo = 6;
// property is not read-only, which is correct because trying to
// define it read-only as above should be an error in the first place
```

### 🙁 Actual behavior

No TypeScript error is shown, neither for `readonly` nor for `declare`.
`readonly` is completely ignored, i.e. writing to the property also does not result in an error.

### 🙂 Expected behavior

In the [auto-accessor PR description](https://github.com/microsoft/TypeScript/pull/49705#issue-1286639012), it is explicitly stated that

> * `accessor` cannot be used with `readonly` or `declare` on the same field declaration.

So I'd expect a corresponding TypeScript error message for both accessor declarations in the example above.
Once using `readonly` on an accessor raises an error, it is correct to ignore this for further analysis, so the current behavior to _not_ report an error when trying to write to the property is correct and should remain as-is.

**Comment by fatcerberus** [comment]
I'm a bit confused why `declare accessor` wouldn't be legal, but `accessor declare` is just weird and looks like an oversight in the parser.

**Comment by RyanCavanaugh** [comment]
Yeah, modifier order is always enforced. This is a bug.

**Comment by fwienber** [comment]
That was fast! 👍
